sor/src/main/java/com/bazaarvoice/emodb/sor/DataStoreModule.java

```diff
@@ -257,10 +257,10 @@ protected void configure() {
     }
 
     @Provides @Singleton
-    Optional<Mutex> provideMutex(DataCenterConfiguration dataCenterConfiguration, @DataStoreZooKeeper CuratorFramework curator) {
+    Optional<TableMutexManager> provideTableMutexManager(DataCenterConfiguration dataCenterConfiguration, @DataStoreZooKeeper CuratorFramework curator) {
         // We only use ZooKeeper if this is the data center that is allowed to edit table metadata (create/drop table)
         if (dataCenterConfiguration.isSystemDataCenter()) {
-            return Optional.<Mutex>of(new CuratorMutex(curator, "/lock/tables"));
+            return Optional.of(new TableMutexManager(curator, "/lock/tables"));
```

Comment:
Building on the above comment, there's an issue of transition period.  During an upgrade a create table request for "newTable1" on the "old" instances will lock on a different lock than a create request for the same table on the new instances.  This could potentially lead to a nasty race condition if two create-table requests are served concurrently by an instance from the "old" and "new" cluster.

There's several possibilities here to address this.  We could use some kind of proxy in front of the old instances to deny access to creating or modifying tables.  Client's would get a 500 error, but at least the system would always be consistent.  Problem is the effort to do so is likely hacky.

My recommendation would be to update this PR so that `TableMutexManager` locks on the partitioned mutex AND the original legacy mutex.  Then in an immediate follow-up release we remove the legacy mutex and only lock on the partitioned mutex.